### DISCUSSION TITLE
Tell `sync` to ignore certain branches

### DISCUSSION THREAD

**Discussion by fingolfin** [arxanas/git-branchless#1117] [Q&A]
Is there a way to instruct the `sync` command to ignore certain branches?

My usecase: I'd like to use `git branchless` in a clone of a repository for which I don't se the rules. And while I am happy to rebase all my own branches, I also need to multiple remote branches. Rebasing those locally is not helpful, I'd like them to keep tracking whatever is on the server.

Is there a way to achieve that?

**Comment by arxanas** [comment]
You can pass a custom revset expression: https://github.com/arxanas/git-branchless/wiki/Command:-git-sync#usage https://github.com/arxanas/git-branchless/wiki/Reference:-Revsets. For example, at my work, I have `git sync` instead aliased to something like `git sync 'draft() % my-branch'`.

**Reply by fingolfin** [reply]
Thank you for the hint. I tried a bit but I couldn't figure out how to restrict it to all commits which have a certain prefix (namely `mh/`). I tried `git branchless sync 'mh/'` and also `git branchless sync 'glob:mh/*'` but both just give an error.

**Reply by arxanas** [reply]
@fingolfin unfortunately, my advice only applies to a statically-known set of branches. There is currently no way in git-branchless to select a set of branches matching a pattern. (Ideally, there would be something like `branches('glob:mh/*')`.)

The `branches()` function currently accepts zero arguments:

https://github.com/arxanas/git-branchless/blob/8ade84e684c49329b24843d4abebc2da2189e2f8/git-branchless-revset/src/builtins.rs#L180

I would accept a PR allowing it to accept an optional pattern.
